lib/async_hooks.js

```diff
@@ -461,6 +461,21 @@ function init(asyncId, type, triggerId, resource) {
   processing_hook = false;
 }
 
+const JSSideProviders = {
+  Timeout: 'timers Timeout',
+  Immediate: 'timers Immediate',
+  TickObject: 'next_tick TickObject'
+};
+
+function _getTypeName(module, type) {
+  return Object.keys(JSSideProviders)
+    .find((k) => JSSideProviders[k] === `${module} ${type}`);
+}
```

Comment:
I have a better idea: lookup by function:
`_getTypeName(setTimeout)`, `_getTypeName(setImmediate)`
that way it's also useful to end users.
The question is will it be robust (no name conflicts)